### ISSUE TITLE
`config set` command writes more than it should

### ISSUE THREAD

**Issue by borland** [OctopusDeploy/cli#71]
Repro on both Windows and Linux

Set `OCTOPUS_HOST` and `OCTOPUS_API_KEY` environment variables to allow you to connect to your server.

Make sure you don't have any pre-existing CLI config file by deleting the file `%APPDATA%\octopus\cli_config.json`

Run `octopus config set space default` (or any other thing that would write a config value)

The CLI will re-generate the config file; open it and inspect the contents

Expected:
config file contains only values that I have explicitly put there using `config set`

```json
{
  "space": "Default"
}
```

Observed:
config file contains all entries, including values picked up from the environment. On my system, it looked like this

```json
{
  "apikey": "API-<redacted>",
  "editor": "notepad",
  "host": "http://localhost:8050",
  "noprompt": false,
  "outputformat": "table",
  "space": "Default"
}
```

**Comment by borland** [comment]
Debugging this, it appears as this is how Viper is designed to work.
The docs are not useful, the best you get from BindEnv is "binds an environment variable..." without any explanation of what "Bind" means.

The values are getting written as part of our call to `config.CreateNewConfig`

Reading the source and stepping through with the debugger:

`viper.SafeWriteConfigAs()` calls (internally) `v.AllKeys()` to get the list of keys to write to the config file.

`AllKeys()` specifically returns keys that were put in v.env by BindEnv
https://github.com/spf13/viper/blob/419fd86e49ef061d0d33f4d1d56d5e2a480df5bb/viper.go#L1969

It then writes values to the config file from all of those keys

**Comment by borland** [comment]
Looks like this was already handled by using a secondary viper inside 'config set'; however the global viper would also write the config file as well, this is how the env vars leaked into it.

Fix: The global viper can pick up all the env vars and flags, but must Not write to the file, and the 'config set' viper can write to the file, but it must NOT pickup env vars and flags.

**Comment by borland** [comment]
Fixed on 0.2.5
